balance.py

```diff
@@ -497,27 +510,67 @@ def callback_teams(self, players, channel):
 
     def cmd_do(self, player, msg, channel):
         """Forces a suggested switch to be done."""
-        if self.suggested_pair:
-            self.execute_suggestion()
+        if not self.suggested_pair:
+            return
+        
+        if not self.switch_opt_in:
+            return
+
+        self.execute_suggestion()
```

Comment:
Player joins on round countdown, script notices that next round switch will be executed, admin wants to execute it right now, types "!do" and says "wtf?".

Not to have this situation, I suggest to return previous version of cmd_do